### PR TITLE
core: modify test_suites_by_environment to provide test count

### PR DESCRIPTION
**Pull request by mwasilew**
Build method test_suites_by_environment will not only provide sorted
list of environments for each build but also test counts
(pass,fail,skip) for each testsuite in each environment

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

File: squad/core/models.py [hunk 1]
```diff
@@ -232,12 +232,22 @@ def test_suites_by_environment(self):
...
         for tr in test_runs:
             for t in tr.tests.all():
-                result[tr.environment].add(t.suite)
+                if t.suite in result[tr.environment].keys():
+                    if t.status in result[tr.environment][t.suite].keys():
+                        result[tr.environment][t.suite][t.status] += 1
+                    else:
+                        result[tr.environment][t.suite].update({t.status: 1})
+                else:
+                    result[tr.environment].update({t.suite: {t.status: 1}})
```
Comment (terceiro) [inline squad/core/models.py]: @you might want to use default values to simplify this. e.g. I think this entire block could be replaced with the following (untested):

```python
result[tr.environment].setdefault(t.suite, { True: 0, False: 0, None: 0 })
result[tr.environment][t.suite][t.status] += 1
```

File: squad/core/models.py [hunk 2]
```diff
@@ -232,12 +232,22 @@ def test_suites_by_environment(self):
...
         for env in result.keys():
-            result[env] = sorted(result[env], key=lambda suite: suite.slug)
+            # there should only be one key in the most nested dict
+            result[env] = sorted(
+                result[env].items(),
+                key=lambda suite_dict: next(iter(suite_dict)).slug)
+
```
Comment (terceiro) [inline squad/core/models.py]: why not just `suite_dict[0].slug` like you did in the tests?

Comment (mwasilew) [reply squad/core/models.py]: I had dict of dicts here and that didn't work (sorting was commented)

File: squad/core/models.py (hunk 1 above)
Comment (mwasilew) [reply squad/core/models.py]: I expect we would get comment that zeros should not be listed in the notification. Probably could be sorted in the email template though.

Comment (mwasilew) [reply squad/core/models.py]: Fixed

File: squad/core/models.py (hunk 2 above)
Comment (mwasilew) [reply squad/core/models.py]: Fixed

**Pull request merged**